x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/expression/predicate/operator/comparison/InsensitiveBinaryComparisonProcessorTests.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.eql.expression.predicate.operator.comparison;
+
+import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
+import org.elasticsearch.common.io.stream.Writeable.Reader;
+import org.elasticsearch.test.AbstractWireSerializingTestCase;
+import org.elasticsearch.xpack.ql.TestUtils;
+import org.elasticsearch.xpack.ql.expression.Literal;
+import org.elasticsearch.xpack.ql.expression.gen.processor.ConstantProcessor;
+import org.elasticsearch.xpack.ql.expression.processor.Processors;
+
+import static org.elasticsearch.xpack.eql.EqlTestUtils.seq;
+import static org.elasticsearch.xpack.eql.EqlTestUtils.sneq;
+import static org.elasticsearch.xpack.ql.tree.Source.EMPTY;
+
+public class InsensitiveBinaryComparisonProcessorTests extends AbstractWireSerializingTestCase<InsensitiveBinaryComparisonProcessor> {
+    public static InsensitiveBinaryComparisonProcessor randomProcessor() {
+        return new InsensitiveBinaryComparisonProcessor(
+            new ConstantProcessor(randomLong()),
+            new ConstantProcessor(randomLong()),
+            randomFrom(InsensitiveBinaryComparisonProcessor.InsensitiveBinaryComparisonOperation.values()));
+    }
+
+    @Override
+    protected InsensitiveBinaryComparisonProcessor createTestInstance() {
+        return randomProcessor();
+    }
+
+    @Override
+    protected Reader<InsensitiveBinaryComparisonProcessor> instanceReader() {
+        return InsensitiveBinaryComparisonProcessor::new;
+    }
+
+    @Override
+    protected NamedWriteableRegistry getNamedWriteableRegistry() {
+        return new NamedWriteableRegistry(Processors.getNamedWriteables());
+    }
+
+    public void testStringEq() {
+        assertEquals(true, p(seq(l("a"), l("a"))));
+        assertEquals(true, p(seq(l("A"), l("a"))));
+        assertEquals(true, p(seq(l("aBcD"), l("AbCd"))));
+        assertEquals(true, p(seq(l("abc"), l("abc"))));
+
+        assertEquals(false, p(seq(l("abc"), l("cba"))));
+    }
+
+    public void testRegularEquals() {
+        assertEquals(true, p(seq(l(12), l(12))));
+        assertEquals(true, p(seq(l(2f), l(2f))));
```

Comment:
Does it make sense to test String against numeric?

Reply:
Not really since the validation would fail. Yet I've added a test for it and put another check inside the operation just in case.